src/overlays/actors/ovl_En_Test/z_en_test.c

```diff
@@ -133,126 +204,1885 @@ static ColliderQuadInit D_808645C8 = {
+                }
+            }
+        }
+
+        if (Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+            checkDist = 200.0f;
+        }
+
+        if (this->actor.xzDistToPlayer <= (80.0f + checkDist)) {
+            Math_SmoothStepToF(&this->unk_7EC, -2.5f, 1.0f, 0.8f, 0.0f);
+        } else if (this->actor.xzDistToPlayer > (110.0f + checkDist)) {
+            Math_SmoothStepToF(&this->unk_7EC, 2.5f, 1.0f, 0.8f, 0.0f);
+        } else {
+            Math_SmoothStepToF(&this->unk_7EC, 0.0f, 1.0f, 6.65f, 0.0f);
+        }
+
+        if (this->unk_7EC != 0.0f) {
+            this->actor.world.pos.x += (Math_SinS(this->actor.shape.rot.y) * this->unk_7EC);
+            this->actor.world.pos.z += (Math_CosS(this->actor.shape.rot.y) * this->unk_7EC);
+        }
+
+        this->skelAnime.playSpeed = this->actor.speedXZ * 0.5f;
+        prevFrame = this->skelAnime.curFrame;
+        SkelAnime_Update(&this->skelAnime);
+
+        temp_f16 = this->skelAnime.curFrame - ABS(this->skelAnime.playSpeed);
+        absPlaySpeed = ABS(this->skelAnime.playSpeed);
+
+        if ((s32)this->skelAnime.curFrame != prevFrame) {
+            temp_v0_2 = (s32)absPlaySpeed + prevFrame;
+
+            if (((temp_v0_2 > 1) && (temp_f16 <= 0)) || ((temp_f16 < 7) && (temp_v0_2 >= 8))) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_WALK);
+            }
+        }
+
+        if ((globalCtx->gameplayFrames & 95) == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_WARAU);
+        }
+
+        yawDiff = playerYaw180 - this->actor.shape.rot.y;
+        yawDiff = ABS(yawDiff);
+
+        if ((yawDiff > 0x6800) || (this->timer == 0)) {
+            EnTest_ChooseAction(this, globalCtx);
+        } else if (this->timer != 0) {
+            this->timer--;
+        }
+    }
+}
+
+void EnTest_SetupSlash1(EnTest* this) {
+    Animation_PlayOnce(&this->skelAnime, &gStalfosDownSlashAnim);
+    func_800F8A44(&this->actor.projectedPos, NA_SE_EN_STAL_WARAU);
+    this->swordCollider.base.atFlags &= ~AT_BOUNCED;
+    this->unk_7C8 = 0x10;
+    this->actor.speedXZ = 0.0f;
+    EnTest_SetupAction(this, EnTest_Slash1);
+    this->swordCollider.info.toucher.damage = 16;
+
+    if (this->unk_7DE != 0) {
+        this->unk_7DE = 3;
+    }
+}
+
+void EnTest_Slash1(EnTest* this, GlobalContext* globalCtx) {
+    this->actor.speedXZ = 0.0f;
+
+    if ((s32)this->skelAnime.curFrame < 4) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0xBB8, 0);
+    }
+
+    if ((s32)this->skelAnime.curFrame == 7) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_SAKEBI);
+    }
+
+    if ((this->skelAnime.curFrame > 7.0f) && (this->skelAnime.curFrame < 11.0f)) {
+        this->swordState = STALFOS_SWORD_ON;
+    } else {
+        this->swordState = STALFOS_SWORD_OFF;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (globalCtx->gameplayFrames % 2) {
```

Comment:
Personal preference.
```suggestion
        if (globalCtx->gameplayFrames % 2 != 0) {
```